public/app/features/admin/LdapPage.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { PureComponent } from 'react';
+import { hot } from 'react-hot-loader';
+import { connect } from 'react-redux';
+import { css } from 'emotion';
+import { NavModel } from '@grafana/data';
+import { FormField } from '@grafana/ui';
+import Page from '../../core/components/Page/Page';
+import { AlertBox } from '../../core/components/AlertBox/AlertBox';
+import { LdapConnectionStatus } from './LdapConnectionStatus';
+import { LdapSyncInfo } from './LdapSyncInfo';
+import config from '../../core/config';
+import { getNavModel } from '../../core/selectors/navModel';
+import { AppNotificationSeverity, LdapError, LdapUser, StoreState, SyncInfo, LdapConnectionInfo } from 'app/types';
+import {
+  loadLdapState,
+  loadLdapSyncStatus,
+  loadUserMapping,
+  clearUserError,
+  clearUserMappingInfo,
+} from './state/actions';
+import { LdapUserInfo } from './LdapUserInfo';
+
+interface Props {
+  navModel: NavModel;
+  ldapConnectionInfo: LdapConnectionInfo;
+  ldapUser: LdapUser;
+  ldapSyncInfo: SyncInfo;
+  ldapError: LdapError;
+  userError?: LdapError;
+
+  loadLdapState: typeof loadLdapState;
+  loadLdapSyncStatus: typeof loadLdapSyncStatus;
+  loadUserMapping: typeof loadUserMapping;
+  clearUserError: typeof clearUserError;
+  clearUserMappingInfo: typeof clearUserMappingInfo;
+}
+
+interface State {
+  isLoading: boolean;
+}
+
+export class LdapPage extends PureComponent<Props, State> {
+  state = {
+    isLoading: true,
+  };
+
+  async componentDidMount() {
+    await this.props.clearUserMappingInfo();
+    await this.fetchLDAPStatus();
+    this.setState({ isLoading: false });
+  }
+
+  async fetchLDAPStatus() {
+    const { loadLdapState, loadLdapSyncStatus } = this.props;
+    return Promise.all([loadLdapState(), loadLdapSyncStatus()]);
+  }
+
+  async fetchUserMapping(username: string) {
+    const { loadUserMapping } = this.props;
+    return await loadUserMapping(username);
+  }
+
+  search = (event: any) => {
+    event.preventDefault();
+    const username = event.target.elements['username'].value;
+    if (username) {
+      this.fetchUserMapping(username);
+    }
+  };
+
+  onClearUserError = () => {
+    this.props.clearUserError();
+  };
+
+  render() {
+    const { ldapUser, userError, ldapSyncInfo, ldapConnectionInfo, navModel } = this.props;
+    const { isLoading } = this.state;
+
+    const tableStyle = css`
```

Comment:
It's possible, will do!

Reply:
As we're mentioned the topic of styles, I thought we should be using `emotion` for styles in all new components? I can see that is not the case in most of the FC components in this PR.

Reply:
@alexanderzobnin regarding css created in render vs outside - you should create styles in render only in case when they depend on the props being passed.

Reply:
@hugohaggmark the `css` template function is from `emotion` isn't it? Or do you mean some other styles?

Reply:
Think it's a mix still as we have some common styles (for tables specifically) that we do not have reusable emotion styled components for. 

Btw regarding were to create styles (in render vs outside). I agree that building emotion class inside render should only be used if class is using props. But if not where is the class built outside? I am not a big fan of global functions that excute when js files are loaded and that is what you get if you define the classes in global scope, or am i wrong, is there another way to create styles lazily on first render?

Reply:
@aocenas I meant the rest of the FC components that aren't using emotion?

Reply:
@torkelo I wasn't precise in my comment. I was thinking about having the styles implementation outside of the render, ie. `const getStyles : () => CSS` function. Css will still be created during the render of course, it's just the render function that's goinna be cleaner.

Reply:
Would love to have this kind of design patterns documented in our FrontEnd guidelines/review @dprokop

Reply:
yea, think could be a good solution